### DISCUSSION TITLE
Community Survey, Development Guidelines and goodbye

### DISCUSSION THREAD

**Discussion by RisingLeaf** [endless-sky/endless-sky#8482] [General]
Goodbye everyone,

As of today I stepping back from the main Endless-Sky repo.

But before I do that, I would like to publish some stuff I have been working on over the last two weeks.

## Survey
The first one being an unofficial survey that I gathered from randomly selected people that are somewhat active.
I did this survey to find out what people are thinking about this project, so here are the results (strongly condensed):

First of all, the positive feedback:
1. Enthusiastic people, fun to work together.
2. Connect with people with the same mindset.
3. Creating the things you like.
4. The cool (and sometimes weird) ideas that people get.
5. How all work together and collaborate.

Problems people have spotted:
on discord:
1. Lack of moderation, too lax moderation (a lot)
2. Too many sibling servers, should be concentrated on the main server.
3. Arguments about play-style are unnecessary.

on development:
1. Constant arguments about small things, over protectiveness of some people.
2. Status of some seems to be too powerful.
3. People hide behind the argument that there should be no limit to their ideas.

Suggestions on how this could be done better:
discord:
1. Stricter moderation
2. An IRL-Matters Channel
3. Explicitly mention the "no bigotry" policy in the rules.
4. More moderators.

development:
1. More transparency from the devs, some things get announced late, when they obviously have been planned for longer.
2. A guide on how to use GitHub.
3. A bounty list for features.
4. More calm discussions than just the usual arguments.
5. Decision making should involve people more.
6. Another Derpy, in the sense that Derpy has a lot of work to do and it is hard to get his attention sometimes.
7. More cooperation between big projects.

Note: I will not provide any information about who said what to respect privacy. In fact I already deleted this info.

## Guidelines
The second thing I worked on are some development guidelines to make it easier to determine if a feature is wanted.
I am not totally happy with where they are right now, but here they are:

```
The ES-Development Guidelines.

In the following the development guidelines for the game Endless Sky
will be stated. These guidelines shall be used to determine if a
specific feature, being art, content, or changes on the code-base,
fits the game.
The guidelines especially should be considered in case of a
disagreement on a feature.

Terms in this document:
1. Leading developer: refers to the person who is managing the
active development and decides what fits the game. Also the person
that publishes releases.

2. Developer: A person with write access to the GitHub repository.

3. Reviewer: A person with triage access to the GitHub repository.

The big guidelines:
1. Goal of the project:
Endless Sky should be a sandbox-style space exploration game, focused
on player driven story telling, with a galaxy filled with alien
species, that all follow a cohesive narrative.

2. The content goal:
A galaxy filled with alien species that follow a cohesive narrative.
Each alien species is supposed to add more variety and perspective
to the game. Additionally several story-line the player can follow to
make the galaxy dynamic and to give insights into the lore behind
an alien species or greater events.
Furthermore the storytelling should not be one sided, but step
away from the typical black and white conflict, to make the game
more diverse.

3. The art goal:
Each alien species should have their own style, a minimum quality
is expected and can be approved by Reviewers or Developer specialized
in art.

4. The engine goal:
The game should be able to run on minimal requirements to make the
game available to a broad player-base. Furthermore the engine aims to
be understandable and every addition has to follow the style
requirements.
The engine tries to feature mechanics that make game-play more
enjoyable and interesting, as well as supporting the content and the
art side of the game. 
Furthermore the engine tries to provide features in a way that they
can easily be used by people creating content for the game or for
mods.

The content guidelines:
1. Everything that gets added to the game needs to have a reason to
exist, be it story telling or additions of other kind, everything
needs to be based on some sort of lore.
    a. Side story lines may also do this by just making the galaxy
    feel more alive.
    b. Alien species are especially big additions, additions of this kind
    need to be based on an especially strong lore and under all
    circumstances need to broaden the perspective of the game.

2. A cohesive narrative is wanted so that two different story lines
give the same feeling to the player.

3. Ships and outfits need to be balanced to fit in with the rest of
the games ships and outfits.

4. Land-able planets need to fit in to the region they are in and
should also serve to deepen the lore of this region.

The art guideline:
// art team, please leave your two cents here.

The code guidelines:
1. Every feature should fit into the genre this game is (Refer
to The goal of this project). Features that are out of place
will not be implemented. The developers know what features fit
the genre and their choice needs to be respected.

2. The code should avoid complexity where possible to lower the barrier
for other contributors.

3. The code should try document what it is doing.

Effectiveness of the guidelines:
1. The guidelines are in effect for the organization Endless Sky,
hosted on GitHub and also for the game Endless Sky, published on
steam. The last one is a product of the first one and therefore
the same rules apply.

2. The leading developer of Endless Sky may cancel out the effect
of the guidelines in cases where they think that the guidelines
are inaccurate.

3. Furthermore if the leading developer has no time or is not active
on the feature, the guidelines may also be canceled out by at least
two developer if there is no other developer opposing it.
// the number may be discussed.

Changing the guidelines:
1. The guidelines should be changed if an inaccuracy is discovered or
if part of the guidelines is not fitting the current development goal.

2. To change the guidelines a discussion is to be opened, with the
reasoning and the proposed changes. This discussion needs to be open
for at least 48 hours, before the proposed change is added to the document.

3. For a change to happen at least two developers need to accept the
proposed change or the leading developer.

4. The leading developer has a veto in changing the guidelines.
```

I wont answer any questions on this post, as I already stated I will step back from this repo.
Greetings
    A Rising Leaf

**Comment by bene-dictator** [comment]
It's really sad to see you go, RisingLeaf. It was nice to work with you when we did, and hopefully your work will lead to bigger things, but you've done a lot for endless-sky. 

It's strange for me, because I feel like some of these issues are ones I don't see going on, as they're in hidden discord channels like #reviewer-conclave (I'm guessing) and things like that. So I'm not really sure what more to say about it apart from what I said in #8396. I can't really say who is more powerful or not because I honestly have no idea what really is going on, and that looks to be true for other contributors too. 

Perhaps some of this is a result of growing very fast as a community-centred project.

**Comment by McloughlinGuy** [comment]
I'm sad to see you go leaf, but I can understand. I enjoyed talking with you and helping you out when I could, and I support your decision. Thank you for being such a great community member.
